bl31/ehf.c

```diff
@@ -408,14 +417,19 @@ static uint64_t ehf_el3_interrupt_handler(uint32_t id, uint32_t flags,
 	 * and shifting the running priority value (platform-supplied).
 	 */
 	idx = pri_to_idx(pri);
+	if (!IS_IDX_VALID(idx)) {
+		ERROR("Invalid index : intr=0x%x req=0x%x idx=0x%x\n",
+				intr, pri, idx);
+		panic();
+	}
 
 	/* Validate priority */
 	assert(pri == IDX_TO_PRI(idx));
 
 	handler = RAW_HANDLER(exception_data.ehf_priorities[idx].ehf_handler);
 	if (!handler) {
-		ERROR("No EL3 exception handler for priority 0x%x\n",
-				IDX_TO_PRI(idx));
+		ERROR("No EL3 exception handler for intr 0x%x priority 0x%x\n",
```

Comment:
PR #1209 suggests that the change is prompted by your receiving unconfigured interrupts. For that case, wouldn't this change (i.e. including the interrupt number) only suffice? Why alther explicit activations and deactivations for debugging output?

Reply:
Yes, the change in function ehf_el3_interrupt_handler() is the main purpose. But all these three functions have called pri_to_idx(), we can change pri_to_idx() function so can let the program to run to here and print out interrupt number.